web3/module.py

```diff
@@ -55,9 +57,43 @@ def apply_result_formatters(
 TReturn = TypeVar("TReturn")
 
 
+@curry
+def retrieve_request_information_for_batching(
+    w3: Union["AsyncWeb3", "Web3"],
+    module: "Module",
+    method: Method[Callable[..., Any]],
+) -> Union[
+    Callable[..., Tuple[Tuple[RPCEndpoint, Any], Sequence[Any]]],
+    Callable[..., Coroutine[Any, Any, Tuple[Tuple[RPCEndpoint, Any], Sequence[Any]]]],
+]:
```

Comment:
Maybe worth creating a type for `RequestInformation` and use that inside the `Callable[...]`s. There are a few places these are used. It could end up something like:
```suggestion
) -> Union[
    Callable[..., RequestInformation],
    Callable[..., Coroutine[Any, Any, RequestInformation],
]:
```